src/track/cueinfo.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+// cueinfo.h
+// Created 2020-02-28 by Jan Holthuis
+
+#include "util/color/color.h"
+
+// DTO for Cue information without dependencies on the actual Track object
+class CueInfo : public QObject {
+    Q_OBJECT
+
+  public:
+    enum class Type {
+        Invalid = 0,
+        HotCue = 1,
+        MainCue = 2,
+        Beat = 3, // unused (what is this for?)
+        Loop = 4,
+        Jump = 5,
+        Intro = 6,
+        Outro = 7,
+        AudibleSound = 8, // range that covers beginning and end of audible
+                          // sound; not shown to user
+    };
+
+    static constexpr double kNoPosition = -1.0;
+    static const int kNoHotCue = -1;
```

Comment:
kNoHotCueNumber

Reply:
constexpr

Should we extract both into the outer scope as `kNoHotCueNumber` and `kNoCuePosition`?

Reply:
Ok. Just to make sure: Using `get/setHotCueNumber` is okay? Right now, `Cue` just uses `getHotCue` but that name looked ambiguous to me.

Reply:
Explicit and consistent

Reply:
> constexpr
> 
> Should we extract both into the outer scope as `kNoHotCueNumber` and `kNoCuePosition`?

Both are also used outside the `CueInfo` and `Cue` classes. Should I still put them into the global namespace?

Reply:
The two constants? Yes, I would suggest it. Use the existing code to check if it feels right. A reality check is worth much more than any theoretical design considerations and discussions.